### DISCUSSION TITLE
Scripts sometimes clicks the verification button, but mostly not with sb.uc_gui_handle_cf

### DISCUSSION THREAD

**Discussion by misterpeople** [seleniumbase/SeleniumBase#2915] [Q&A] [answered]
Hey, so I'm trying to bypass Cloudflare verification and CLICK the button to verify with the scripts below, and I do not understand why most of the time, even when the Chrome page is on top, it does not even detect the button and it does not click on it. I cannot understand why it's so random and what I am doing wrong. I tried it on a different script (which is below as well), but yet again, sometimes it detects it and prints logging.info("CAPTCHA iframe found"), and most of the time it prints logging.info("No CAPTCHA iframe found, assuming automatic verification"), but we are on the CAPTCHA page.

Both scripts WORK and click the button Id say 40% of the time AND I do not understand why its not 100%.

Both scripts:
  ```
from seleniumbase import SB
import logging

# Configure logging
logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')

#Proxy Configuration
PROXY_HOST= "geo.demo.com"
PROXY_PORT= "demo"
PROXY_USER= "demo"
PROXY_PASS= "demo"



# Construct the proxy URL
proxy_url = f"{PROXY_USER}:{PROXY_PASS}@{PROXY_HOST}:{PROXY_PORT}"

with SB(uc=True, proxy=proxy_url) as sb:
    url = "https://www.virtualmanager.com/en/login"
    sb.uc_open_with_reconnect(url, 5)
    print(sb.get_page_title())
    sb.uc_gui_handle_cf()  # Ready if needed!
    print(sb.get_page_title())
    sb.wait_for_element('input[name*="email"]', timeout=100)
    sb.assert_element('input[name*="email"]')

    sb.wait_for_element('input[name*="login"]', timeout=100)
    sb.assert_element('input[name*="login"]')

    sb.set_messenger_theme(location="bottom_center")
    sb.post_message("SeleniumBase wasn't detected!")
    ```
    

```
script 2: 
```
def click_verify_human(sb, chat_id, bot):
    try:
        # Wait for the CAPTCHA iframe to be visible
        max_wait = 40
        start_time = time.time()
        while time.time() - start_time < max_wait:
            if sb.is_element_present('iframe[title*="challenge"]'):
                logging.info("CAPTCHA iframe found")
                bot.send_message(chat_id, "CAPTCHA challenge detected")

                # Use uc_gui_handle_cf() to handle the CAPTCHA
                sb.uc_gui_handle_cf("iframe[title*='challenge']")
                logging.info("Used uc_gui_handle_cf() to handle CAPTCHA")
                bot.send_message(chat_id, "Attempted to solve CAPTCHA using new method")
                return True
            time.sleep(1)

        logging.info("No CAPTCHA iframe found, assuming automatic verification")
        bot.send_message(chat_id, "No CAPTCHA challenge detected, proceeding")
        return True

    except Exception as e:
        logging.error(f"Error handling CAPTCHA challenge: {e}")
        bot.send_message(chat_id, f"Error handling CAPTCHA challenge: {e}")
        return False

   ```

```

**Comment by mdmintz** [comment, marked as answer]
It's probably timing-related.
You could try the newer `uc_gui_click_cf()` command in `4.28.4` (and newer versions).

```python
from seleniumbase import SB

with SB(uc=True, test=True) as sb:
    url = "https://www.virtualmanager.com/en/login"
    sb.uc_open_with_reconnect(url, 4)
    sb.uc_gui_click_cf()
    sb.assert_element('input[name*="email"]')
    sb.assert_element('input[name*="login"]')
    sb.set_messenger_theme(location="bottom_center")
    sb.post_message("SeleniumBase wasn't detected!")
```

**Reply by misterpeople** [reply]
i used the same script u sent, while the cloudflare captcha shows "verify" its clicking on it without waiting for the box to even load, so it still misses most of the time

**Reply by mdmintz** [reply]
The you need to increase the `reconnect_time`.
Eg. `sb.uc_open_with_reconnect(url, 8)`

**Reply by misterpeople** [reply]
I did and it didnt work well so I did with (url, 13) which works a bit better but still misses a lot, I tried to play with it but seems like nothing is actually working right its either missing or not even clicking. Is there any work around that can be implemented to actually wait for the exact element we need to click on to load and only then clicking on it? or anything else?

**Reply by mdmintz** [reply]
If you wait for it, you get detected, so you have to guess. Guess higher than you need.

**Reply by misterpeople** [reply]
I see so basically guessing higher should click on it even if we wait for it to load 20 seconds for example, but sometimes it says that it clicked but actually does nothing, any idea why?

**Reply by mdmintz** [reply]
Not sure. Experiment until you figure it out.

**Reply by misterpeople** [reply]
I see, I will. thanks! do you have any code example of re-try clicking?
